src/vm/variables.rs

```diff
@@ -1,13 +1,47 @@
-// This currently is just serving as a place-holder for reserved variable
-//   names. 
+use vm::types::Value;
+use vm::contexts::{LocalContext, Environment};
+use vm::errors::{Error, ErrType, InterpreterResult as Result};
 
-pub const TX_SENDER: &str = "tx-sender";
+pub enum NativeVariables {
+    TxSender, BlockHeight, BurnBlockHeight
+}
 
-static RESERVED_VARIABLES: &[&str] = 
-    &[TX_SENDER,
-      "block-height",
-      "burn-block-height"];
+impl NativeVariables {
+    pub fn lookup_by_name(name: &str) -> Option<NativeVariables> {
+        use vm::variables::NativeVariables::*;
```

Comment:
Why is this line necessary if the enum is declared above?

Reply:
Enum variants aren't brought into scope by default.

Reply:
D'oh! You're right.